### PR TITLE
refactor(circuits): remove duplicate constraints

### LINKED ISSUE #305: Duplicate constraints on process message circuit
The circuit which process messages contains a duplicate check for the constraint, `enoughVoiceCredits` which is initially validated in the `messageValidator` circuit. 

Therefore this should be removed when processing each message since we assume the message is already valid, having the appropriate number of voice credits

### PR DESCRIPTION
**Pull request by ctrlc03**
# Description

Remove duplicate constraints on circuits by re-using signals and removing unnecessary checks.

## Related issue(s)

fix #305

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: circuits/circom/utils/non-qv/messageValidator.circom [hunk 1]
```diff
@@ -46,6 +46,10 @@ template MessageValidatorNonQv() {
 
     // True when the command is valid; otherwise false.
     signal output isValid;
+    // True if the state leaf index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/non-qv/messageValidator.circom:49]: I think it should be `The state leaf index is valid or not` or `True if the state leaf index is valid` ?
I mean just choose either `true` or `or not` not both?

File: circuits/circom/utils/non-qv/messageValidator.circom [hunk 2]
```diff
@@ -46,6 +46,10 @@ template MessageValidatorNonQv() {
...
+    signal output isStateLeafIndexValid;
+    // True if the vote option index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/non-qv/messageValidator.circom:51]: same here

File: circuits/circom/utils/non-qv/stateLeafAndBallotTransformer.circom [hunk 3]
```diff
@@ -66,9 +66,13 @@ template StateLeafAndBallotTransformerNonQv() {
     
     // True when the command is valid; otherwise false.
     signal output isValid;
+    // True if the state leaf index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/non-qv/stateLeafAndBallotTransformer.circom:69]: same here

File: circuits/circom/utils/non-qv/stateLeafAndBallotTransformer.circom [hunk 4]
```diff
@@ -66,9 +66,13 @@ template StateLeafAndBallotTransformerNonQv() {
...
+    signal output isStateLeafIndexValid;
+    // True if the vote option index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/non-qv/stateLeafAndBallotTransformer.circom:71]: same here

File: circuits/circom/utils/qv/messageValidator.circom [hunk 5]
```diff
@@ -46,6 +46,10 @@ template MessageValidator() {
 
     // True when the command is valid; otherwise false.
     signal output isValid;
+    // True if the state leaf index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/qv/messageValidator.circom:49]: same here

File: circuits/circom/utils/qv/messageValidator.circom [hunk 6]
```diff
@@ -46,6 +46,10 @@ template MessageValidator() {
...
+    signal output isStateLeafIndexValid;
+    // True if the vote option index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/qv/messageValidator.circom:51]: same here

File: circuits/circom/utils/qv/stateLeafAndBallotTransformer.circom [hunk 7]
```diff
@@ -66,9 +66,13 @@ template StateLeafAndBallotTransformer() {
     
     // True when the command is valid; otherwise false.
     signal output isValid;
+    // True if the state leaf index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/qv/stateLeafAndBallotTransformer.circom:69]: same here

File: circuits/circom/utils/qv/stateLeafAndBallotTransformer.circom [hunk 8]
```diff
@@ -66,9 +66,13 @@ template StateLeafAndBallotTransformer() {
...
+    signal output isStateLeafIndexValid;
+    // True if the vote option index is valid or not
```
Comment (kittybest) [inline circuits/circom/utils/qv/stateLeafAndBallotTransformer.circom:71]: same here

**Review by kittybest — COMMENTED** [review]
LGTM, just left some comments

**Review by 0xmad — APPROVED** [review]
@ctrlc03 thanks!

**Pull request merged**